examples/prediction/xgboost-train-and-deploy-in-gcp.ipynb

```diff
@@ -0,0 +1,300 @@
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.preprocessing import Imputer\n",
+    "from xgboost import XGBRegressor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logging.basicConfig(format='%(message)s')\n",
+    "logging.getLogger().setLevel(logging.INFO)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def read_input(file_name, test_size=0.25):\n",
+    "    \"\"\"Read input data and split it into train and test.\"\"\"\n",
+    "    data = pd.read_csv(file_name)\n",
+    "    data.dropna(axis=0, subset=['SalePrice'], inplace=True)\n",
+    "\n",
+    "    y = data.SalePrice\n",
+    "    X = data.drop(['SalePrice'], axis=1).select_dtypes(exclude=['object'])\n",
+    "\n",
+    "    train_X, test_X, train_y, test_y = train_test_split(X.values,\n",
+    "                                                      y.values,\n",
+    "                                                      test_size=test_size,\n",
+    "                                                      shuffle=False)\n",
+    "\n",
+    "    imputer = Imputer()\n",
+    "    train_X = imputer.fit_transform(train_X)\n",
+    "    test_X = imputer.transform(test_X)\n",
+    "\n",
+    "    return (train_X, train_y), (test_X, test_y)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_model(train_X,\n",
+    "                train_y,\n",
+    "                test_X,\n",
+    "                test_y,\n",
+    "                n_estimators,\n",
+    "                learning_rate):\n",
+    "    \"\"\"Train the model using XGBRegressor.\"\"\"\n",
+    "    model = XGBRegressor(n_estimators=n_estimators, learning_rate=learning_rate)\n",
+    "\n",
+    "    model.fit(train_X,\n",
+    "            train_y,\n",
+    "            early_stopping_rounds=40,\n",
+    "            eval_set=[(test_X, test_y)])\n",
+    "\n",
+    "    print(\"Best RMSE on eval: %.2f with %d rounds\",\n",
+    "               model.best_score,\n",
+    "               model.best_iteration+1)\n",
+    "    return model\n",
+    "\n",
+    "def eval_model(model, test_X, test_y):\n",
+    "    \"\"\"Evaluate the model performance.\"\"\"\n",
+    "    predictions = model.predict(test_X)\n",
+    "    logging.info(\"mean_absolute_error=%.2f\", mean_absolute_error(predictions, test_y))\n",
+    "\n",
+    "def save_model(model, model_file):\n",
+    "    \"\"\"Save XGBoost model for serving.\"\"\"\n",
+    "    joblib.dump(model, model_file)\n",
+    "    logging.info(\"Model export success: %s\", model_file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def gcs_copy(src_path, dst_path):\n",
```

Comment:
why not just make save_model upload to GCS? And maybe use the client library instead of shelling out?

Reply:
Wanted to call out separately so that users can comment it out if they want. Have no issue in moving it to save_mode(). Using gsutil here because I had issues in using google-storage python client. Since this is not an in-tree feature I didn't want to spend time debugging.